### PR TITLE
Upgrade Version and add Unit Tests

### PR DESCRIPTION
**Pull request by Pieeer1**
- Adds Unit Tests using GDUnit4Mono
- Upgrades Version to Godot 4.2.1

### CODE REVIEW

File: gdunit4_testadapter/TestAdapterRunner.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+namespace GdUnit4.TestAdapter;
+
+public partial class TestAdapterRunner : Api.TestRunner
```
Comment (MikeSchulze) [inline gdunit4_testadapter/TestAdapterRunner.cs:3]: Do not add the `TestAdapterRunner` files to your repository!
These files are generated and the folder `gdunit_testadapter` should be added to the .gitignore file

Comment (GeorgeS2019) [reply gdunit4_testadapter/TestAdapterRunner.cs:3]: > the folder gdunit_testadapter should be added to the .gitignore file

@MikeSchulze  Can this be automated?

Comment (Pieeer1) [reply gdunit4_testadapter/TestAdapterRunner.cs:3]: @MikeSchulze I am automating it in a pipeline, I need it initially for the automated pipeline test run. I am going to remove it if the pipeline tests can run without it

Comment (MikeSchulze) [reply gdunit4_testadapter/TestAdapterRunner.cs:3]: This class and scene is auto generated when the folder not exists.
The test adapter is generating this files on the fly

Comment (Pieeer1) [reply gdunit4_testadapter/TestAdapterRunner.cs:3]: This has been resolved in https://github.com/Pieeer1/GodotVoipNet/pull/13

**Pull request merged**